### PR TITLE
added CustomGlobalExceptionHandler and validation in Dto

### PR DESCRIPTION
**Pull request by ArtemReizer**

### CODE REVIEW

**Review by IhorKalaur — APPROVED** [review]
Everything works, but I would add informativeness to the message, and set a limit on the length of characters.

File: src/main/java/com/app/bookstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.validator.constraints.ISBN;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
```
Comment (vovasalyha) [inline src/main/java/com/app/bookstore/dto/CreateBookRequestDto.java:14]: Add length (size) constraints to not catch an exception on repository level

File: src/main/java/com/app/bookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+    @NotNull
     private String author;
```
Comment (vovasalyha) [inline src/main/java/com/app/bookstore/dto/CreateBookRequestDto.java:17]: same here

File: src/main/java/com/app/bookstore/exceptions/CustomGlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode statusCode,
+            WebRequest webRequest
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
```
Comment (vovasalyha) [inline src/main/java/com/app/bookstore/exceptions/CustomGlobalExceptionHandler.java]: better create a class for error response body (or use some classes created by spring)

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Good. Let's improve it a bit.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**